internal/provider/resources/concurrency_limit.go

```diff
@@ -0,0 +1,263 @@
+package resources
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/google/uuid"
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/int64planmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/planmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/stringplanmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/prefecthq/terraform-provider-prefect/internal/api"
+	"github.com/prefecthq/terraform-provider-prefect/internal/provider/customtypes"
+	"github.com/prefecthq/terraform-provider-prefect/internal/provider/helpers"
+)
+
+var (
+	_ = resource.ResourceWithConfigure(&ConcurrencyLimitResource{})
+	_ = resource.ResourceWithImportState(&ConcurrencyLimitResource{})
+)
+
+// ConcurrencyLimitResource contains state for the resource.
+type ConcurrencyLimitResource struct {
+	client api.PrefectClient
+}
+
+// ConcurrencyLimitResourceModel defines the Terraform resource model.
+type ConcurrencyLimitResourceModel struct {
+	BaseModel
+
+	AccountID   customtypes.UUIDValue `tfsdk:"account_id"`
+	WorkspaceID customtypes.UUIDValue `tfsdk:"workspace_id"`
+
+	Tag              types.String `tfsdk:"tag"`
+	ConcurrencyLimit types.Int64  `tfsdk:"concurrency_limit"`
+}
+
+// NewConcurrencyLimitResource returns a new ConcurrencyLimitResource.
+//
+//nolint:ireturn // required by Terraform API
+func NewConcurrencyLimitResource() resource.Resource {
+	return &ConcurrencyLimitResource{}
+}
+
+// Metadata returns the resource type name.
+func (r *ConcurrencyLimitResource) Metadata(_ context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_concurrency_limit"
+}
+
+// Configure initializes runtime state for the resource.
+func (r *ConcurrencyLimitResource) Configure(_ context.Context, req resource.ConfigureRequest, resp *resource.ConfigureResponse) {
+	if req.ProviderData == nil {
+		return
+	}
+
+	client, ok := req.ProviderData.(api.PrefectClient)
+	if !ok {
+		resp.Diagnostics.Append(helpers.ConfigureTypeErrorDiagnostic("resource", req.ProviderData))
+
+		return
+	}
+
+	r.client = client
+}
+
+// Schema defines the schema for the resource.
+func (r *ConcurrencyLimitResource) Schema(_ context.Context, _ resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		Description: "The resource `concurrency_limit` represents a concurrency limit for a tag.",
```

Comment:
```suggestion
		Description: "The resource `concurrency_limit` represents a concurrency limit. Concurrency limits allow you to manage execution efficiently, controlling how many tasks, flows, or other operations can run simultaneously. They are ideal for optimizing resource usage, preventing bottlenecks, and customizing task execution.",
```

I wasn't sure "for a tag" was totally right, so I grabbed this description from https://docs.prefect.io/v3/develop/global-concurrency-limits#manage-global-concurrency-and-rate-limits - open to tweaking it